config/e2e/monitoring.yaml

```diff
@@ -0,0 +1,711 @@
+- apiGroups:
+  - apm.k8s.elastic.co
+  resources:
+  - apmservers
+  - apmservers/status
+  - apmservers/finalizers
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - enterprisesearch.k8s.elastic.co
+  resources:
+  - enterprisesearches
+  - enterprisesearches/status
+  - enterprisesearches/finalizers
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - beat.k8s.elastic.co
+  resources:
+  - beats
+  - beats/status
+  - beats/finalizers
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - admissionregistration.k8s.io
+  resources:
+  - mutatingwebhookconfigurations
+  - validatingwebhookconfigurations
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: monitoring-elastic-operator
+  labels:
+    test-run: {{ .TestRun }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: monitoring-elastic-operator
+subjects:
+- kind: ServiceAccount
+  name: monitoring-elastic-operator
+  namespace: {{ .E2ENamespace }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: monitoring
+  namespace: {{ .E2ENamespace }}
+  labels:
+    control-plane: monitoring
+    test-run: {{ .TestRun }}
+spec:
+  selector:
+    matchLabels:
+      control-plane: monitoring
+  serviceName: monitoring
+  template:
+    metadata:
+      annotations:
+        co.elastic.metrics/metricsets: collector
+        co.elastic.metrics/module: prometheus
+        co.elastic.metrics/hosts: '${data.host}:9090'
+        co.elastic.metrics/period: 10s
+        # if monitoring secrets are specified, filebeat is deployed and provides the below configuration
+        {{ if .MonitoringSecrets }}
+        # Rename the fields "error" to "error.message" and "source" to "event.source"
+        # This is to avoid a conflict with the ECS "error" and "source" documents.
+        co.elastic.logs/raw: "[{\"type\":\"container\",\"json.keys_under_root\":true,\"paths\":[\"/var/log/containers/*${data.kubernetes.container.id}.log\"],\"processors\":[{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"error\",\"to\":\"_error\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"_error\",\"to\":\"error.message\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"source\",\"to\":\"_source\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"_source\",\"to\":\"event.source\"}]}}]}]"
+        {{end}}
+      labels:
+        control-plane: monitoring
+        test-run: {{ .TestRun }}
+    spec:
+      serviceAccountName: monitoring-elastic-operator
+      containers:
+      - image: docker.elastic.co/eck-snapshots/eck-operator:1.3.0-SNAPSHOT-2020-09-23-199f4aab
```

Comment:
This is temporary as one of the fixes from `1.3` (modifying init containers) is needed here.